app/scripts/views/marketing_snippet.js

```diff
@@ -60,8 +64,20 @@ define(function (require, exports, module) {
   const View = BaseView.extend({
     template: Template,
 
+    /**
+     * Initialize.
+     *
+     * @param {Object} [options=}|]
+     *  @param {String} [options.marketingId] - marketing ID to use for logging.
```

Comment:
Many of the changes to marketing_snippet.js could be extracted into its own PR to simplify review. For example, the marketingId and which options could be extracted, but unused, against master.